src/utils/filter.ts

```diff
@@ -10,36 +11,33 @@ import { sanitizedValue } from './sanitize.js';
  * @param query the incoming QueryBuilder.
  * @returns The modified QueryBuilder respecting the provided filters.
  */
-export function filterSelectQuery<DB = unknown, TB extends keyof DB = keyof DB, O = unknown>(
+export function filterSelectQuery<DB = DwnDatabaseType, TB extends keyof DB = keyof DB, O = unknown>(
   filters: Filter[],
   query: SelectQueryBuilder<DB, TB, O>
 ): SelectQueryBuilder<DB, TB, O> {
-  return query.where((eb) => {
-    // we are building multiple OR queries out of each individual filter.
-    const or: OperandExpression<SqlBool>[] = [];
-    for (let filter of filters) {
-      // processFilter will take a single filter adding it to the query to be evaluated as an OR operation with the other filters.
-      or.push(processFilter(eb, filter));
-    }
-    // Evaluate the array of expressions as an OR operation.
-    return eb.or(or);
-  });
+  const extractedFilters = extractTagsAndSanitizeFilters(filters);
+
+  return query.where((eb) =>
+    eb.or(extractedFilters.map(({ filter, tag }) =>
+      processFilterWithTag(eb, filter, tag)
+    ))
+  );
 }
 
 /**
  * Returns an array of OperandExpressions for a single filter.
  * Each property within the filter is evaluated as an AND operand,
  * if a property has an array of values it will treat it as a OneOf (IN) within the overall AND query.
- * This way each Filer has to be a complete match, but the collection of filters can be evaluated as chosen.
+ * This way each Filter has to be a complete match, but the collection of filters can be evaluated as chosen.
  *
  * @param eb The ExpressionBuilder from the query.
  * @param filter The filter to be evaluated.
- * @returns An array of OperandExpressions to be evaluated by the caller.
+ * @returns An array of OperandExpression<SqlBool> to be evaluated by the caller.
  */
-function processFilter<DB = unknown, TB extends keyof DB = keyof DB>(
+function processFilter<DB = DwnDatabaseType, TB extends keyof DB = keyof DB>(
```

Comment:
Discussion post my PR: do you think it is possible to consolidate the tags version and non-tags version into one method? At a glance it seems like the tags is a super/complex set?

Reply:
Opened an issue for this to look at in a separate effort/PR:
https://github.com/TBD54566975/dwn-sql-store/issues/29